controller/controller_server/utils.py

```diff
@@ -8,43 +10,161 @@
 import controller.controller_server.messages as messages
 from controller.array_action.config import FC_CONNECTIVITY_TYPE, ISCSI_CONNECTIVITY_TYPE
 from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.controller_types import ArrayConnectionInfo, ObjectIdInfo, ObjectParameters
 from controller.controller_server.errors import ObjectIdError, ValidationException
 from controller.csi_general import csi_pb2
 
 logger = get_stdout_logger()
 
 
-def get_array_connection_info_from_secret(secrets):
+def _parse_raw_json(raw_json):
+    try:
+        parsed_json = json.loads(raw_json)
+    except json.decoder.JSONDecodeError:
+        raise ValidationException(messages.invalid_json_parameter_message.format(raw_json))
```

Comment:
could you include the error message in our exception message here?

Reply:
sure, done